src/backend/distributed/master/worker_node_manager.c

```diff
@@ -401,6 +403,55 @@ NodeIsPrimaryWorker(WorkerNode *node)
 }
 
 
+/*
+ * CanUseCoordinatorLocalTablesWithReferenceTables returns true if we
+ * are allowed to use coordinator local tables with reference tables
+ * for joining them.
+ */
+bool
+CanUseCoordinatorLocalTablesWithReferenceTables(void)
+{
+	/*
+	 * Using local tables of coordinator with reference tables is only allowed
+	 * if we are in the coordinator.
+	 */
+	if (!IsCoordinator())
+	{
+		return false;
+	}
+
+	/*
+	 * Check if coordinator can have reference table placements (i.e added
+	 * to pg_dist_node previously)
+	 */
+	if (!CanHaveReferenceTablePlacements())
+	{
+		return false;
+	}
+
+	return true;
+}
+
+
+/*
+ * CanHaveReferenceTablePlacements returns true if current node can have
+ * reference table placements.
+ */
+static bool
+CanHaveReferenceTablePlacements()
```

Comment:
Maybe CanHaveReferenceTable? I think it is simpler and has the same meaning.

Reply:
But there the point is if the current node have local placements for reference tables

Reply:
If a node can have reference tables it will have reference table placements. To me having placements is a detail. Is it possible that a node can have reference tables but not its placements? My question is more like `if reference tables are not replicated to the coordinator, would we say that coordinator has reference tables?`

Well this is kind of brainstorming :D Feel free to keep it like this. I think your point makes sense too.

Reply:
>  it possible that a node can have reference tables but not its placements? 

yes, if we did not add coordinator to pg_dist_node, it normally does not include local reference table placements

> if reference tables are not replicated to the coordinator, would we say that coordinator has reference tables?

I guess yes 😄

Reply:
> yes, if we did not add coordinator to pg_dist_node, it normally does not include local reference table placements

my point was that in this case I would say no it doesnt have reference tables :D but I am okay to keep it like this :)